tests/validate-dirty-db.py

```diff
@@ -0,0 +1,130 @@
+                    action='store_true')
+parser.add_argument("--keep-logs", help="Don't delete var/lib/node_* folders upon test completion",
+                    action='store_true')
+
+args = parser.parse_args()
+debug=args.v
+pnodes=1
+topo="mesh"
+delay=1
+chainSyncStrategyStr=testUtils.Utils.SyncResyncTag
+total_nodes = pnodes
+killCount=1
+killSignal=testUtils.Utils.SigKillTag
+
+killEosInstances= not args.dont_kill
+dumpErrorDetails=args.dump_error_details
+keepLogs=args.keep_logs
+
+seed=1
+testUtils.Utils.Debug=debug
+testSuccessful=False
+
+random.seed(seed) # Use a fixed seed for repeatability.
+cluster=testUtils.Cluster(walletd=True)
+
+try:
+    cluster.setChainStrategy(chainSyncStrategyStr)
+
+    cluster.killall()
+    cluster.cleanup()
+
+    Print ("producing nodes: %d, topology: %s, delay between nodes launch(seconds): %d, chain sync strategy: %s" % (
+        pnodes, topo, delay, chainSyncStrategyStr))
+
+    Print("Stand up cluster")
+    if cluster.launch(pnodes, total_nodes, topo=topo, delay=delay, dontBootstrap=True) is False:
+        errorExit("Failed to stand up eos cluster.")
+
+    node=cluster.getNode(0)
+    if node is None:
+        errorExit("Cluster in bad state, received None node")
+
+    Print("Kill cluster nodes.")
+    cluster.killall()
+
+    def runNodeosAndGetOutput(nodeId, waitTime=3):
+        """Startup nodeos, wait for waitTime (before forced shutdown) and collect output. Stdout, stderr and return code are returned in a dictionary.
+        Large waittimes have potential to deadlock as the popen PIPE buffer can get filled up."""
+        Print("Launching nodeos process id: %d" % (nodeId))
+        dataDir="var/lib/node_%02d" % (nodeId)
+        cmd="programs/nodeos/nodeos --config-dir etc/eosio/node_bios --data-dir var/lib/node_bios"
+        Print("cmd: %s" % (cmd))
+        proc=subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        try:
+            proc.wait(waitTime)
+        except (subprocess.TimeoutExpired) as _:
+            Print("ERROR: Nodeos is running beyond the defined wait time. Hard killing nodeos instance.")
+            proc.send_signal(signal.SIGKILL)
+            return (False, None)
+
+        output={}
+        output["stdout"] = proc.stdout.read().decode("utf-8")
+        output["stderr"] = proc.stderr.read().decode("utf-8")
+        output["returncode"] = proc.returncode
+        return (True, output)
+    
+    Print("Restart nodeos repeatedly to ensure dirty database flag sticks.")
+    nodeId=0
+    waitTime=3
+    
+    for i in range(0,3):
+        Print("Attempt %d." % (i))
+        ret = runNodeosAndGetOutput(nodeId, waitTime)
+        if not ret or not ret[0]:
+            exit(1)
+
+        #Print(ret)
+
+        stderr=ret[1]["stderr"]
+        retCode=ret[1]["returncode"]
+        assert(retCode == 2)
+        assert("database dirty flag set" in stderr)
+
+    testSuccessful=True
+finally:
```

Comment:
This may not be something to do as part of this, but can we make this finally block a common method that we can call for all the tests?